support/metrics/metricsTracker.go

```diff
@@ -286,3 +319,28 @@ func (mt *MetricsTracker) sendEvent(eventType string, eventProps interface{}) er
 	}
 	return nil
 }
+
+func (mt *MetricsTracker) mergeEventProps(eventProps interface{}) (map[string]interface{}, error) {
```

Comment:
this function needs a test

if it's easier to write the test, this can be a function that's not on `metricsTracker` and is a standalone function. This can have the signature of `(parentProps map[string]interface{}, additionalProps map[string]interface{}) (map[string]interface{}, error)`

It may even be good to put this in `utils/functions.go` along with the test in `utils/functions_test.go` calling it `mergeMaps` or something. Your call on best approach.

Reply:
Post-rewrite, is a unit test still worthwhile? Now, it just merges two maps into one - it iterates over each and assigns to a new map. I've written the test, but not sure it's still needed.